### PR TITLE
Shopping Cart application

### PR DESCRIPTION
**Pull request by psolarvi**

### CODE REVIEW

File: shopping-cart/src/test/java/com/redhat/shopping/integration/white/box/ShoppingCartTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        assertEquals(0, this.cartService.totalItems());
+
+        // Adding a product to the cart with a specified initial quantity
+        this.cartService.addProduct(existingProductId, initialQty);
+
+        // Adding again the same product to the cart with a specified quantity
+        this.cartService.addProduct(existingProductId, incrementalQty);
+
+        // Assert: the total number of items in the cart is the sum of the initial and the incremental quantities
+        assertEquals(initialQty + incrementalQty, this.cartService.totalItems());
```
Comment (jramcast) [inline shopping-cart/src/test/java/com/redhat/shopping/integration/white/box/ShoppingCartTest.java:69]: The assert is clearer by if we just hardcode the expected value:

```suggestion
        assertEquals(110, this.cartService.totalItems());
```

File: shopping-cart/src/test/java/com/redhat/shopping/integration/white/box/ShoppingCartTest.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        int incrementalQty    = 100;
+
+        // Assert: there are no items in the cart
+        assertEquals(0, this.cartService.totalItems());
+
+        // Adding a product to the cart with a specified initial quantity
+        this.cartService.addProduct(existingProductId, initialQty);
+
+        // Adding again the same product to the cart with a specified quantity
+        this.cartService.addProduct(existingProductId, incrementalQty);
```
Comment (jramcast) [inline shopping-cart/src/test/java/com/redhat/shopping/integration/white/box/ShoppingCartTest.java:66]: I would also say that the test setup is clearer if we just hardcode the parameter values in the `addProduct` calls

**Review by jramcast — CHANGES_REQUESTED** [review]
Pretty complete app to show how integration testing works :)

I left some comments. Many of them are just suggestions we can discuss.

Also, this looks like the final solution, so instead of merging into `master`, we should merge this  pr to the `solutions` branch

File: shopping-cart/src/main/java/com/redhat/shopping/cart/AddToCartCommand.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package com.redhat.shopping.cart;
+
+public class AddToCartCommand {
+    public int id;
+    public int qty;
+
+    public AddToCartCommand() {
+    }
```
[the remark this answers is not in the document]
Comment (psolarvi) [reply shopping-cart/src/main/java/com/redhat/shopping/cart/AddToCartCommand.java:8]: That constructor is required by quarkus/java in order to do some magic and transform the request JSON body into an object.

File: shopping-cart/src/main/java/com/redhat/shopping/cart/CartView.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+import java.util.ArrayList;
+import java.util.Collection;
+
+public class CartView {
+
+    public ArrayList<CartItem> products;
+    public int totalItems;
+
+    public CartView() {
+    }
```
[the remark this answers is not in the document]
Comment (psolarvi) [reply shopping-cart/src/main/java/com/redhat/shopping/cart/CartView.java:12]: Is the same reason, quarkus/java forced me to add that constructor. Without it, the app explodes when I try to return a view of the cart. :confused:

**Review by jramcast — APPROVED** [review]

**Pull request merged**